components/raftstore/src/store/worker/pd.rs

```diff
@@ -1207,13 +1214,17 @@ where
         }
 
         stats = collect_report_read_peer_stats(HOTSPOT_REPORT_CAPACITY, report_peers, stats);
-        let (capacity, used_size, available) = match collect_engine_size(
-            &self.coprocessor_host,
-            Some(&store_info),
-            self.snap_mgr.get_total_snap_size().unwrap(),
-        ) {
-            Some((capacity, used_size, available)) => (capacity, used_size, available),
-            None => return,
+        let (capacity, used_size, available) = if store_info.is_some() {
```

Comment:
```suggestion
        let (capacity, used_size, available) = if let Some(store_info) = store_info {
```

Reply:
IMO, `is_some` is better?

Reply:
I think minghua means from his way, the var `store_info` will be unboxed for the following usage, which could be more convenient

Reply:
Got it.
But we need the original wrapped `store_info` to make a further judgement whether the given task is a fake `StoreHeartbeat` generated by `force_report_slow_score` function, I revert the modification here to the original one => still use `is_some()`.

This can be reviewed in:
https://github.com/LykxSassinator/tikv/blob/41204d9748236f8a2d7d3d53d381b5fb1d68a9a6/components/raftstore/src/store/worker/pd.rs#L1281